projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/withProxy.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+const cache = {};
+const cacheResponse = ( requestId, blob, freshness ) => {
+	// Cache at most 100 items
+	const cacheKeys = Object.keys( cache );
+	if ( cacheKeys.length > 100 ) {
+		delete cache[ cacheKeys[ 0 ] ];
+	}
+
+	cache[ requestId ] = blob;
+
+	// Self-remove this entry after `freshness` ms
+	setTimeout( () => {
+		delete cache[ requestId ];
+	}, freshness );
+};
+
+const withProxy = WrappedComponent => props => {
+	const { proxyIsEnabled, proxySiteIdOrSlug, proxyMediaPath, proxyQuery, ...restProps } = props;
+
+	const [ imageObjectUrl, setImageObjectUrl ] = useState( '' );
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ isError, setIsError ] = useState( false );
+
+	const requestId = 'tiled-gallery-proxied-image-';
+	useEffect( () => {
+		if ( ! proxyIsEnabled || imageObjectUrl || isFetching || isError ) {
+			return;
+		}
+		if ( cache[ requestId ] ) {
+			// Load image from cache
+			const url = URL.createObjectURL( cache[ requestId ] );
+			setImageObjectUrl( url );
+			console.log( 'got image from cache', { url } );
+		} else {
+			// Not in cache, send a request
+			setIsFetching( true );
+			const safeQuery = ( proxyQuery || '' ).replace( /^\?/, '' );
+			/*
+			apiFetch( {
+				//path: `/sites/${ proxySiteIdOrSlug }/atomic-auth-proxy/file?path=${ proxyMediaPath }&${ safeQuery }`,
+				//path: `https://public-api.wordpress.com/sites/${ proxySiteIdOrSlug }/atomic-auth-proxy/file?path=${ proxyMediaPath }&${ safeQuery }`,
+				apiNamespace: 'wpcom/v2',
+			} )
+            */
+			//testsitemmrtag.files.wordpress.com/2021/03/drone-4.jpg?resize=750x750
+			// https://flarypod.jurassic.tube/wp-content/uploads/2021/03/drone-2-1.jpg
+			// fetch(
+			// 	'https://public-api.wordpress.com/wpcom/v2/sites/testsitemmrtag.wordpress.com/atomic-auth-proxy/file/wp-content/uploads/2021/03/drone-4.jpg?resize=214%2C214'
+			// )
+			fetch(
+				'https://public-api.wordpress.com/wpcom/v2/sites/flarypod.jurassic.tube/atomic-auth-proxy/file/wp-content/uploads/2021/03/drone-2-1.jpg?resize=214%2C214'
```

Comment:
I think this is the wrong format; the file path should be a query param, shouldn't it?

Reply:
https://github.com/Automattic/wp-calypso/blob/d80ea4c187645a954776ec0afd9a9193dc06cbdf/client/lib/wpcom-undocumented/lib/undocumented.js#L2576

Reply:
correct - this is working for me https://public-api.wordpress.com/wpcom/v2/sites/mmrtt1.wpcomstaging.com/atomic-auth-proxy/file?path=/wp-content/uploads/2020/09/qi-bin-w4hbafegiac-unsplash.jpg&ssl=1&resize=219%2C219&_envelope=1

Reply:
also, putting any jurassic tube urls here is a bad idea, and will make testing complicated.. even though we can easily trick our jetpack sites into thinking they're atomic, it will probably be best to test the api when pointing it at real atomic sites

Reply:
Using JT, I can see the blobs being loaded, but no actual calls to the auth proxy endpoint – is that expected?

Reply:
As of 28d82504b, I was trying to use `wpcomProxyRequest()` to send off requests, but it wasn't working. It sets up an iframe, and sends messages to that iframe, and the iframe is supposed to listen to those messages and send the actual requests. However, something wasn't working in the process and I wasn't able to figure out what. I have removed that and as of a4c048e56dcce2, am trying to use a jetpack endpoint, which connects to a wpcom endpoint, however, that is also running into issues of its own (which I'll talk about in the main thread)